wasmtime/_func.py

```diff
@@ -91,11 +91,15 @@ def __call__(self, *params: IntoVal) -> Union[IntoVal, Sequence[IntoVal], None]:
 
         ty = self.type
         param_tys = ty.params
+        if len(params) > len(param_tys):
+            raise WasmtimeError("too many parameters provided")
+        if len(params) < len(param_tys):
+            raise WasmtimeError("too few parameters provided")
+
+        param_vals = [Val._convert(ty, params[i], "keep alive across call")
```

Comment:
FWIW there's *tons* of instances where we don't keep something alive across a call, but I'm led to believe that's ok in Python because reference counting doesn't happen until the end of a scope. In that sense I don't think that we need to manually set an owner here to keep things alive across the call.